### ISSUE TITLE
Mesh pipelines are invalidating dynamic states it does not have

### ISSUE THREAD

**Issue by ziga-lunarg** [KhronosGroup/Vulkan-ValidationLayers#8378]
Mesh pipelines can't have some of the dynamic states that graphics pipeline has, such as `VK_DYNAMIC_STATE_PRIMITIVE_RESTART_ENABLE_EXT `. But currently binding a mesh pipeline will invalidate all dynamic states, including those that are not part of the mesh pipeline. 

An example to reproduce this bug would be:

1. Set `vkCmdSetPrimitiveRestartEnable()`
2. Bind a mesh pipeline
3. Bind graphics pipeline that has `VK_DYNAMIC_STATE_PRIMITIVE_RESTART_ENABLE_EXT` dynamic state
4. Draw

**Comment by spencer-lunarg** [comment]
Waiting on confirmation this is invalid or not

https://gitlab.khronos.org/vulkan/vulkan/-/issues/3996

```c++
TEST_F(PositiveDynamicState, MeshPipelineInvalidtion) {
    TEST_DESCRIPTION("https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/8378");
    SetTargetApiVersion(VK_API_VERSION_1_2);
    AddRequiredExtensions(VK_EXT_EXTENDED_DYNAMIC_STATE_2_EXTENSION_NAME);
    AddRequiredExtensions(VK_EXT_MESH_SHADER_EXTENSION_NAME);
    AddRequiredFeature(vkt::Feature::extendedDynamicState2);
    AddRequiredFeature(vkt::Feature::meshShader);
    RETURN_IF_SKIP(Init());
    InitRenderTarget();

    CreatePipelineHelper pipe(*this);
    pipe.AddDynamicState(VK_DYNAMIC_STATE_PRIMITIVE_RESTART_ENABLE_EXT);
    pipe.CreateGraphicsPipeline();

    VkShaderObj ms(this, kMeshMinimalGlsl, VK_SHADER_STAGE_MESH_BIT_EXT, SPV_ENV_VULKAN_1_2);
    CreatePipelineHelper mesh_pipe(*this);
    mesh_pipe.shader_stages_ = {ms.GetStageCreateInfo(), pipe.fs_->GetStageCreateInfo()};
    mesh_pipe.CreateGraphicsPipeline();

    m_command_buffer.Begin();
    m_command_buffer.BeginRenderPass(m_renderPassBeginInfo);

    vk::CmdSetPrimitiveRestartEnableEXT(m_command_buffer, VK_TRUE);
    vk::CmdBindPipeline(m_command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, mesh_pipe);
    vk::CmdBindPipeline(m_command_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipe);
    vk::CmdDraw(m_command_buffer, 3, 1, 0, 0);

    m_command_buffer.EndRenderPass();
    m_command_buffer.End();
}
```

**Comment by ziga-lunarg** [comment]
I've realized this was once an issues already in #5551

**Comment by spencer-lunarg** [comment]
1. confirmed the spec does allow this

<img width="995" height="94" alt="Image" src="https://github.com/user-attachments/assets/d0a8bfed-b551-4bbd-913b-b5da397de084" />

2. I just need to build a list/map for each dynamic state that is tied to each stage ... 😢
